tests/D2L.CodeStyle.Analyzers.Test/Common/MutabilityInspectorTests.cs

```diff
@@ -67,120 +67,190 @@ public class MutabilityInspectorTests {
 		}
 
 		[Test]
-		public void IsTypeMutable_PrimitiveType_False() {
+		public void InspectType_PrimitiveType_NotMutable() {
 			var type = Field( "uint foo" ).Type;
+			var expected = MutabilityInspectionResult.NotMutable();
 
-			Assert.IsFalse( m_inspector.IsTypeMutable( type ) );
+			var actual = m_inspector.InspectType( type );
+
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_NullablePrimitiveType_False() {
+		public void InspectType_NullablePrimitiveType_NotMutable() {
 			var type = Field( "uint? foo" ).Type;
+			var expected = MutabilityInspectionResult.NotMutable();
+
+			var actual = m_inspector.InspectType( type );
 
-			Assert.IsFalse( m_inspector.IsTypeMutable( type ) );
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_NullableNonPrimitiveType_True() {
+		public void InspectType_NullableNonPrimitiveType_NotMutable() {

```

Comment:
I changed this test because (1) the behaviour of deep checking is already tested by other tests and (2) this test actually really cares about making sure that _nullable_ structs are inspected (`System.Nullable` is itself a struct).